specification/purview/Azure.Analytics.Purview.DataMap/models.tsp

```diff
@@ -0,0 +1,2146 @@
+<li>If a list of entityTypes are supplied, where one inherits
+from another, this will be rejected. This should encourage cleaner
+classificationsDefs</li>
+</ul>
+""")
+  entityTypes?: string[];
+
+  @doc("An array of sub types.")
+  subTypes?: string[];
+
+  @doc("An array of super types.")
+  superTypes?: string[];
+
+  @doc("An array of relationship attributes.")
+  relationshipAttributeDefs?: AtlasRelationshipAttributeDef[];
+
+  @doc("The default value.")
+  defaultValue?: string;
+
+  @doc("An array of enum element definitions.")
+  elementDefs?: AtlasEnumElementDef[];
+
+  @doc("""
+The relationshipEndDef represents an end of the relationship. The end of the
+relationship is defined by a type, an
+attribute name, cardinality and whether
+it  is the container end of the relationship.
+""")
+  endDef1?: AtlasRelationshipEndDef;
+
+  @doc("""
+The relationshipEndDef represents an end of the relationship. The end of the
+relationship is defined by a type, an
+attribute name, cardinality and whether
+it  is the container end of the relationship.
+""")
+  endDef2?: AtlasRelationshipEndDef;
+
+  @doc("""
+The Relationship category determines the style of relationship around
+containment and lifecycle.
+UML terminology is used for the
+values.
+<p>
+ASSOCIATION is a relationship with no containment. <br>
+COMPOSITION
+and AGGREGATION are containment relationships.
+<p>
+The difference being in the
+lifecycles of the container and its children. In the COMPOSITION case,
+the
+children cannot exist without the container. For AGGREGATION, the life
+cycles
+of the container and children are totally independent.
+""")
+  relationshipCategory?: RelationshipCategory;
+
+  @doc("The label of the relationship.")
+  relationshipLabel?: string;
+
+  @doc("An array of attribute definitions.")
+  attributeDefs?: AtlasAttributeDef[];
+}
+
+@doc("The definitions of types.")
+model AtlasTypesDef {
+  @doc("businessMetadataDefs")
+  businessMetadataDefs?: AtlasBusinessMetadataDef[];
+
+  @doc("An array of classification definitions.")
+  classificationDefs?: AtlasClassificationDef[];
+
+  @doc("An array of entity definitions.")
+  entityDefs?: AtlasEntityDef[];
+
+  @doc("An array of enum definitions.")
+  enumDefs?: AtlasEnumDef[];
+
+  @doc("An array of relationship definitions.")
+  relationshipDefs?: AtlasRelationshipDef[];
+
+  @doc("An array of struct definitions.")
+  structDefs?: AtlasStructDef[];
+
+  @doc("An array of term template definitions.")
+  termTemplateDefs?: TermTemplateDef[];
+}
+
+@doc("Term template definition for glossary term.")
+model TermTemplateDef {
+  ...AtlasStructDef;
+}
+
+@doc("The basic information of the type definition.")
+model AtlasTypeDefHeader {
+  @doc("The enum of type category.")
+  category?: TypeCategory;
+
+  @doc("The GUID of the type definition.")
+  guid?: string;
+
+  @doc("The name of the type definition.")
+  name?: string;
+}
+
+@discriminator("kind")
+@doc("Base model for search filter")
+model SearchFilter {
```

Comment:
It's a bit awkward that this model has no properties. @johanste is this an acceptable design for a base model for discriminated types?

Reply:
Comment a little bit. We want to define a base model as SearchFilter which can be a filter for typeName, guid, andFilter, orFilter etc. As there are no shared properties for search filter, this model does not contain any properties.